cmd/operator/main.go

```diff
@@ -99,6 +99,21 @@ func (n namespaces) asSlice() []string {
 	return ns
 }
 
+// Helper function for checking CRD prerequisites
+func checkPrerequisites(ctx context.Context, logger log.Logger, cc *k8sutil.CRDChecker, allowedNamespaces []string, verbs map[string][]string, groupVersion, resourceName string) (bool, error) {
+	err := cc.CheckPrerequisites(ctx, allowedNamespaces, verbs, groupVersion, resourceName)
+	switch {
+	case errors.Is(err, k8sutil.ErrPrerequiresitesFailed):
+		level.Warn(logger).Log("msg", fmt.Sprintf("%s CRD not supported because prerequisites are not met", resourceName), "err", err)
```

Comment:
```suggestion
		level.Warn(logger).Log("msg", fmt.Sprintf("%s CRD not supported", resourceName), "reason", err)
```

this would log as

````
msg="prometheusagents CRD not supported" reason="unmet prerequisites: prometheusagents resource (apiGroup: \"monitoring.coreos.com/v1alpha1\") not installed"
````